test/system/src/main/java/io/pravega/test/system/framework/TestExecutorFactory.java

```diff
@@ -9,21 +9,29 @@
  */
 package io.pravega.test.system.framework;
 
+import lombok.Getter;
 import org.apache.commons.lang3.NotImplementedException;
 
 public class TestExecutorFactory {
-    private static final TestExecutor MARATHON_SEQUENTIAL_EXECUTOR = new RemoteSequential();
+    @Getter(lazy = true)
+    private final TestExecutor marathonSequentialExecutor = new RemoteSequential();
+
+    @Getter(lazy = true)
+    private final TestExecutor dockerExecutor = new DockerRemoteSequential();
 
     public enum TestExecutorType {
         LOCAL,
+        DOCKER,
         REMOTE_SEQUENTIAL,
         REMOTE_DISTRIBUTED //TODO: Yet to be implemented.
```

Comment:
I know this is not changed here, but we need an issue for this TODO.

Reply:
done.